Dark mode filters/rules/transactions

carkom opened the pull request:
Trying to split darkTheme PR into smaller bits (#922).

It's actually really hard to find a good place to splice the work. Much of the smaller elements overlap so if you change one thing it changes another. And it's in this many-to-many relationship so you can't do one without doing another.

Anyway here is where I landed for the first PR. It's able to largely keep most things the same on any elements/pages I'm not touching while changing what I can for Filters/Rules.

Transactions Tables are all included, along with Modals, etc.

This one will be the largest since it is encorporating much of the base items. Hopefully we can find a way to split future PRs to be a bit smaller.

MatissJanis commented:
👋 Thanks for helping with darkmode! In order not to unnecessarily spend my & other reviewer time: would you mind making sure the CI checks are all green before sending the PR for review? Looks like currently a couple of jobs are failing due to various issues.

MatissJanis reviewed (commented):
👋 I started reviewing this PR, but I'm worried that it will stall again because there are quite a lot of changes for light-mode. Would it be possible to do only the darkmode changes while keeping light-mode the same? I'm not too fussed about how darkmode looks like now as it is under a feature flag, so we can continue iterating and improving it. But light-mode needs to be stable.

Also: I wouldn't expect to see font-size/width/height changes here. We'd only need to swap colors when switching between the two modes.

Some differences I'm observing:
- [x] significantly different scrollbars
- [x] rules edit modal is quite different
- [x] filters dont have pill background anymore
- [x] tables have a dropped shadow now
- [x] schedules modal has some weird colors
- [x] "repeats" field in schedules modal is broken

Sorry to push back again. I really do want to see this merged, but I also don't want to risk breaking the product.

carkom commented:
Okay @MatissJanis, ready for a check. I fixed all issues you mentioned above. I've tried to keep everything as consistent as possible with light mode.

Problem is (and I'm sure bio can attest to this) - there's a lot of inconsistency. Eg. some errorText colors are r4 others are r5, some Tooltips are slightly different colors than other (same with menus/pills/buttons/etc.). As a result of all this I've kept everything largely the same except for small things like slight shading differences (eg. keeping just on errorText color or making all the menus have the same background color, etc.).

I think there is more "consistency work" to be done but it doesn't make sense to jump into that until the theme code is completely ported. Once all the theme stuff is done we can work on removing a lot of the "alt" color schemes and getting everything the same.

Have a look, let me know what you'd like to tweak and we can do it.

MatissJanis commented:
Some more differences I'm seeing:
- [ ] schedules & rules modal - input boxes have gray background
- [ ] scrollbar in sidenav is still very different from the original
- [ ] setting page - "formatting" select boxes for some reason have different dimensions
- [ ] setting page - reset sync button has a different color
- [ ] budget - table headings no longer have gray background
- [ ] budget (screenshot)
<img width="59" alt="Screenshot 2023-08-03 at 16 48 11" src="https://github.com/actualbudget/actual/assets/886567/783e48b9-190b-4a57-934f-1c29238656f3">
- [ ] budget - clicking on the primary budget number opens a tooltip - this now has a different background (gray vs white)
- [ ] budget - focusing on input fields now has a blue border instead of purple

Would you mind checking these things yourself too? What I do is open the old & new version and then just switch between the two tabs. The changes become very easily visible then.

carkom commented:
>     * [ ]  scrollbar in sidenav is still very different from the original

This seems fine for me in Chrome but off in Firefox. Maybe a browser issue?
>     * [ ]  budget (screenshot)
> <img alt="Screenshot 2023-08-03 at 16 48 11" width="59" src="https://user-images.githubusercontent.com/886567/258166272-783e48b9-190b-4a57-934f-1c29238656f3.png">

This seems to be an issue in the live build. This is my live server (v.23.8.0). Also please note the header has a transparent background so it overlays other elements as you scroll down.
![image](https://github.com/actualbudget/actual/assets/55785687/2919f029-ee42-498c-aa9f-dc69bc50572e)

MatissJanis commented:
> * [ ]  scrollbar in sidenav is still very different from the original
> This seems fine for me in Chrome but off in Firefox. Maybe a browser issue?

I'll try to check this and let you know. But there is definitely some change in here that has made it different from master branch.

> * [ ]  budget (screenshot)
> This seems to be an issue in the live build. This is my live server (v.23.8.0). Also please note the header has a transparent background so it overlays other elements as you scroll down.

I'm comparing with https://demo.actualbudget.org/ (master) - the issue is not there for me. Is it there for you?

As for the background: I should have clarified. This is what I meant. It's no longer gray here, but instead is white.

<img width="753" alt="Screenshot 2023-08-03 at 21 35 40" src="https://github.com/actualbudget/actual/assets/886567/e6a4092e-d9e2-444f-8fec-9742a64285a9">

MatissJanis commented:
Fixed the scrollbars.

carkom commented:
> I'm comparing with https://demo.actualbudget.org/ (master) - the issue is not there for me. Is it there for you?

Yes it's there for me on the demo page. But only in Firefox. Chrome is fine. I may open a new PR to try to sniff this one out. I'd rather not complicate this PR any more than neccessary.

carkom commented:
Okay, I fixed all the issues @MatissJanis pointed out from above. I've also gone through myself (eg. changed some table coloring issues, etc.). Thoughts?

MatissJanis commented on packages/desktop-client/src/components/forms.tsx:
```diff
@@ -106,14 +108,14 @@ export const Checkbox = (props: CheckboxProps) => {
           },
         ],
-        props.style,
+        props.styles,
```
Thanks!

carkom replied:
Thanks @biohzrddd!

MatissJanis commented on packages/desktop-client/src/components/transactions/TransactionsTable.js:
```diff
@@ -975,6 +970,7 @@ const Transaction = memo(function Transaction(props) {
       )}
       {(() => (
         <PayeeCell
+          /* Payee field for all transactions */
```
Neat! Thanks for adding these comments

carkom replied:
credit to @biohzrddd for this.

MatissJanis reviewed (commented):
Code looks good. But I found a few very small issues. Hopefully fixing them won't be super hard.

- [ ] checkboxes now have a purple border
- [ ] schedules - "add specific days" background color is wrong (should be white)
- [ ] schedules - the fields that appear after clicking "add specific days" have a weird extra border around them + size seems wrong

- <img width="36" alt="Screenshot 2023-08-05 at 10 06 10" src="https://github.com/actualbudget/actual/assets/886567/ef758e48-bc28-48e7-a227-87437e082120">
- <img width="256" alt="Screenshot 2023-08-05 at 10 07 27" src="https://github.com/actualbudget/actual/assets/886567/2d8fe3ed-bf94-47e5-8775-f93ed90ce93a">
- <img width="315" alt="Screenshot 2023-08-05 at 10 07 52" src="https://github.com/actualbudget/actual/assets/886567/4a0616db-521d-47de-8be5-1118a6cfb1d9">

carkom commented:
> Code looks good. But I found a few very small issues. Hopefully fixing them won't be super hard.

No issue
>     * [ ]  checkboxes now have a purple border
>     * [ ]  schedules - "add specific days" background color is wrong (should be white)
>     * [ ]  schedules - the fields that appear after clicking "add specific days" have a weird extra border around them + size seems wrong

All fixed!

MatissJanis reviewed (approved):
Looks good! Thanks for all your hard work!

(I won't merge just yet.. we're planing on doing a v23.8.1 patch release very soon. Once that is out - I'll merge this PR).

carkom commented:
> Looks good! Thanks for all your hard work!
> 
> (I won't merge just yet.. we're planing on doing a v23.8.1 patch release very soon. Once that is out - I'll merge this PR).

Thanks! Hopefully this is the hardest one. The rest should be eaiser and require less time to review going forward.
